AprilSylph/XKit-Rewritten
# feat(Postblock): Make options UI items links if possible

#### User-facing changes
- Postblock's options UI links to posts you have blocked, so you can tell what they actually are if you want to unblock them. Posts that have not been seen since this upgrade won't have links, since we didn't store the required information previously.
- When viewing a single blocked post in /blog/view, there is a visual indicator with an unhide and unblock button instead of, you know, nothing. 

#### Technical explanation

A map* of post ID -> blog UUID entries is stored as a separate storage entry, rather than migrating the blocked posts list to e.g. an array of `[id, ?uuid]`. You could definitely do the latter, but that kind of thing requires a bit of thinking to ensure you correctly migrate old entries and don't have race conditions.

*I unapologetically call POJO records "maps." Sue me.

#### Issues this closes
discussion #402

### Testing steps

- Load a saved set of preference data with blocked posts. Confirm that they function correctly but are not links in the options UI.
- Block a post. Confirm that it is a link in the options UI, and that the link opens the post in question when clicked.
- Confirm that blocked posts have a new UI when viewed alone, that the new UI looks correct both with and without the "use blog colors when viewing blogs" option, and that its buttons function.
- Navigate to a post that was blocked before switching to this code. Confirm that it becomes a link.
